CMakeLists.txt

```diff
@@ -48,19 +48,153 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
+  string(REPLACE "\\" "/" BUILDENV_BASEPATH "${BUILDENV_BASEPATH}")
+  string(REPLACE "\\" "/" CMAKE_TOOLCHAIN_FILE "${CMAKE_TOOLCHAIN_FILE}")
+endif()
+
+if(
+  WIN32
+  AND
+    (
+      NOT DEFINED CMAKE_TOOLCHAIN_FILE
+      OR
+        (
+          DEFINED BUILDENV_BASEPATH
+          AND DEFINED BUILDENV_NAME
+          AND
+            NOT
+              CMAKE_TOOLCHAIN_FILE
+                MATCHES
+                "${BUILDENV_BASEPATH}/${BUILDENV_NAME}/scripts/buildsystems/vcpkg.cmake"
+        )
+    )
+)
```

Comment:
```suggestion
if(
  DEFINED CMAKE_TOOLCHAIN_FILE
  AND DEFINED MIXXX_VCPKG_ROOT
  AND
    NOT
      CLEAN_CMAKE_TOOLCHAIN_FILE
        MATCHES
        "${MIXXX_VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
)
  message(
    FATAL_ERROR
    "Build environment: '${MIXXX_VCPKG_ROOT}' does not match the environment used previously: '${CMAKE_TOOLCHAIN_FILE}'. Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory."
  )
endif()

if(WIN32 AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
  if(NOT DEFINED BUILDENV_BASEPATH)
```

Reply:
The suggested condition did not work, as it disabled the buildenv download in the normal workflow (because windows_buildenv.bat sets CMAKE_TOOLCHAIN_FILE). I adjusted the condition for the error message accordingly.